apps/myjobs/app/models/workflow_file.rb

```diff
@@ -0,0 +1,55 @@
+class WorkflowFile
+  attr_reader :path, :staged_dir
+
+  def initialize(path, staged_dir)
+    @path = path
+    @staged_dir = staged_dir
+  end
+
+  # Get an array of files of a directory meeting the criteria for job scripts in alphabetical order
+  #
+  # Filter the files by file size, extension and content
+  def suggested_script?
+    @suggested_script ||= (valid_size? && ( has_suggested_extensions? || starts_with_shebang_line? || has_specific_command?))
+  end
+ 
+  # Return true if the file meets the basic requirements to be a job script
+  def valid_script?
+    @valid_script ||= valid_size?
+  end
+  
+  # Return true if file name extention is in the recommended extension list
+  def has_suggested_extensions?
+    suggested_job_script_file_extensions = [".sh", ".job", ".slurm", ".batch", ".qsub", ".sbatch", ".srun", ".bsub"]
+    suggested_job_script_file_extensions.include? File.extname(path)
+  end
+  
+  # Return true if file starts with a shebang line
+  def starts_with_shebang_line?
+    begin
+      (File.open(path) { |f| f.read(2) }) == "#!"
+    rescue
+      false
+    end
+  end
+
+  # Return true if first 1000 bytes of file contain '#PBS' or '#SBATCH"
+  def has_specific_command?
```

Comment:
Better to rename `has_specific_command?` to `has_resource_manager_directive?`. Also based on https://slurm.schedmd.com/rosetta.pdf these are all the script directives we should search for:

- `#PBS`
- `#SBATCH`
- `#BSUB`
- `#$`

And maybe open an issue in ood_core to add to job adapters a method to return the "script_directive_prefix" which will return these strings. Because this is an example of something that should ultimately be handled by ood_core.